exec/memory/base/src/main/java/io/netty/buffer/DrillBuf.java

```diff
@@ -431,12 +482,18 @@ public boolean equals(Object obj) {
   }
 
   @Override
-  public synchronized ByteBuf retain(int increment) {
-    if(rootBuffer){
-      this.rootRefCnt.addAndGet(increment);
-    }else{
-      b.retain(increment);
+  public ByteBuf retain(int increment) {
```

Comment:
how is this retain used compared to the other one (that takes an allocator)? Is this one only used when in the same allocator?